libbeat/processors/add_id/docs/add_id.asciidoc

```diff
@@ -0,0 +1,18 @@
+[[add_id]]
+=== Generate an ID for an event
+
+The `add_id` processor generates a unique ID for an event.
+
+[source,yaml]
+-----------------------------------------------------
+processors:
+ - add_id: ~
+-----------------------------------------------------
+
+The following settings are supported:
+
+`target_field`:: (Optional) Field where the generated ID will be stored. Default is `@metadata.id`.
```

Comment:
somewhere we should document that the Elasticsearch output will use `@metadata.id` as document ID if present. Plus we should document on how users can use the field if sending via Logstash.

Reply:
Good idea. I was initially thinking of doing this in a blog post (that would cover the introduction of this processor and the `fingerprint` processor, from a use case perspective). But I think it's useful to have this in our docs somewhere as well, since users will be looking at those more frequently over time.

I wonder if this sort of documentation belongs in the ES output and LS output docs, since it's strictly agnostic to _what_ generated the ID (e.g. this processor, but could be something else too). @dedemorton WDYT?

Reply:
@ycombinator I'm planning to work on a related issue this week: https://github.com/elastic/beats/issues/13739. We should talk about your plans for the blog post and see what makes sense for the docs. I'd like to avoid burying the content too deeply under the config settings. I think we need a topic that describes what the document ID is (with links to the ES docs), how it's set, and why you want to set it (deduplication etc). Maybe we can chat later this week.